### PR TITLE
Reorganize plugins into different models

### PR DESCRIPTION
**Pull request by andermi**
reorganized models with plugins

- `mbari_wec_base`: raw model with links and joints
- `mbari_wec`: add multiphysics plugins for air spring and EH-PTO
- `mbari_wec_ros`: add ros2 controllers

inheritance:
`mbari_wec_base` -> `mbari_wec` -> `mbari_wec_ros`

`worlds/mbari_wec.sdf` uses `mbari_wec_ros` and adds the joint_state stuff

### CODE REVIEW

File: buoy_description/models/mbari_wec_ros/model.config [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <email>louise@openrobotics.org</email>
+  </author>
+  <author>
+    <name>Dharini Dutia</name>
+    <email>dharini@openrobotics.org</email>
+  </author>
+
+  <description>
+    ROS2-enabled MBARI-WEC (Wave Energy Converter) model for ocean sensing and harvesting wave energy
+    with quasi-static multi-physics plugins for Pneumatic Spring and Electro-Hydraulic Power Take-Off.
```
Comment (chapulina) [inline buoy_description/models/mbari_wec_ros/model.config:22]: I think it's worth mentioning that the plugins necessary to get TF, `robot_state_publisher` and odometry from ROS aren't included.

**Review by chapulina — COMMENTED** [review]
I haven't tried running it but the idea LGTM 👍🏽

File: buoy_description/models/mbari_wec_ros/model.config (hunk 1 above)
Comment (andermi) [reply buoy_description/models/mbari_wec_ros/model.config:22]: added
- `JointStatePublisher`
- `PosePublisher`
- `OdometryPublisher`

**Review by quarkytale — APPROVED** [review]
Works for me

**Pull request merged**